src/components/ui/password-input/password-input.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { UseFormReturn } from 'react-hook-form';
+import { InputValidationError } from '../input-validation-error';
+
+type Props = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  form: UseFormReturn<any, any, any>;
+  isConfirm?: boolean;
+  isForgot?: boolean;
+};
+
+export const PasswordInput = ({ isConfirm, isForgot, form }: Props) => {
+  const {
+    register,
+    getValues,
+    formState: { errors },
+  } = form;
+
+  const validatePassword = () =>
+    isConfirm ? undefined : /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9]).{6,15}$/;
+
+  const validateConfirmPassword = () =>
+    isConfirm ? () => getValues('password') === getValues('confirm_password') : undefined;
```

Comment:
use constants for password and confirm_password, it will avoid some typo errors when u are using it more then once

Reply:
created const enum for form dields names